djangocms_blog/cms_wizards.py

```diff
@@ -43,6 +48,27 @@ def save(self, commit=True):
         self.instance._set_default_author(get_current_user())
         return super(PostWizardForm, self).save(commit)
 
+    def clean_slug(self):
+        """
+        Generate a valid slug, in case the given one is taken
+        """
+        source = self.cleaned_data.get('slug', '')
+        lang_choice = self.language_code
+        if not source:
+            source = slugify(self.cleaned_data.get('title', ''))
+        qs = Post._default_manager.namespace(
+            self.cleaned_data.get('app_config').namespace
+        ).active_translations(lang_choice).language(lang_choice)
+        used = list(qs.values_list('translations__slug', flat=True))
+
+        slug = source
+        i = 1
+        if used:
```

Comment:
Maybe we could send some text via messages or similar, to show clearly to the user that the slug is generated as an alternative string because the original one is taken. Not so necessary anyway.

Reply:
Unfortunately we are completely outside the request-response cycle here.
Could you open an issue to track this. If we can come up with a solution for your suggestion, that would a super nice UX

Reply:
I see, ok let's keep track of it inside an issue!